File: projects/06/clojure/src/array.clj
;;;; array.clj creates an array, reads values, and prints it back out 
;;;;
;;;; Input: size of the array and the value(s) to be store in the array
;;;; Output: the value(s) in the array
;;;;
;;;; Completed by: David Choi
;;;; Date: 3-11-2014
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns array)      ; name the program

(defn printArray [aArray]
	(doseq [item aArray]
		(println item)
	)
)

(defn readArray [aArray aSize]
	(loop [i 0]
		(when (< i aSize)
			(print "Enter in value for array: ")(flush)
			(let [value(read)]
			(aset-float aArray i value)
			(recur(inc i)))
		)
	)
)


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; main function
;;; Input: a, b, c
;;; Output: quadratic roots of given values
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
	(println "Enter the size of the array: ")(flush)
	(let
		[aSize(read)
		aArray(make-array Float/TYPE aSize)]
		aArray(readArray aArray aSize)
		(printArray aArray)
		) 
)

File: projects/06/array.java
/* array.java creates an array, reads values, and prints it back out 
 * Completed by: David Choi
 ***************************************************************/

import java.io.*;
import java.util.Scanner;

// java class array
public class array
{
  // main program
  public static void main(String[] args){
  	Scanner keyboard = new Scanner(System.in);

  	System.out.println("\nEnter the size of the array: ");

  	int size = Integer.parseInt(keyboard.next());
  	double[] aArray = new double[size];

  	aArray = readArray(aArray, aArray.length, keyboard);
  	printArray(aArray);
  }
    

  public static double[] printArray(double[] aArray){
  	for(int i = 0; i < aArray.length; i++)
  		System.out.print(aArray[i] + "\n");
  	return aArray;
  }

  public static double[] readArray(double aArray[], int size, Scanner keyboard){
  	System.out.println("\nEnter the values for the array: ");
  	for(int i = 0; i < size; i ++)
  		aArray[i] = Double.parseDouble(keyboard.next());
  	return aArray;
  }
}

